satellite/satellitedb/repairqueue.go

```diff
@@ -24,12 +30,57 @@ func (r *repairQueue) Insert(ctx context.Context, seg *pb.InjuredSegment) error
 	return err
 }
 
-func (r *repairQueue) Select(ctx context.Context) (seg *pb.InjuredSegment, err error) {
+func (r *repairQueue) postgresSelect(ctx context.Context) (seg *pb.InjuredSegment, err error) {
 	//todo :  add or age > some time (1 hour?)
-	err = r.db.QueryRowContext(ctx, `SELECT data FROM injuredsegments ORDER BY id LIMIT 1 WHERE attempted IS NOT NULL`).Scan(&seg)
+	err = r.db.QueryRowContext(ctx, r.db.Rebind(`
+	UPDATE injuredsegments SET attempted = ? WHERE path = (
+		SELECT path FROM injuredsegments
+		WHERE attempted IS NULL
+		ORDER BY path FOR UPDATE SKIP LOCKED LIMIT 1
+	) RETURNING data`), time.Now().UTC()).Scan(&seg)
```

Comment:
If I'm not wrong, this shouldn't do what's expected, because a pointer to `*pb.InjuredSegment` and it should be a non-nil pointer to `pb.InjuredSegment`

Reply:
hmm I didn't totally understand this comment-- I know that this function should return a `*pb.InjuredSegment`, but you're saying that `pb.InjuredSegment` should be returned instead?

Reply:
Yeah, its not immediately clear that would work, given that seg starts out as a nil pointer.  However, I believe it does work.  We use a Scanner for InjuredSegment which calls proto.Unmarshal() which calls injuredSegment,Reset(), which makes seg non-nil.

https://github.com/storj/storj/blob/904d009a732964c795518eaf1fbea1f2c580614f/pkg/pb/scannerValuer.go#L23

https://github.com/storj/storj/blob/904d009a732964c795518eaf1fbea1f2c580614f/pkg/pb/datarepair.pb.go#L32